### PR TITLE
(SDK-269) Add acceptance tests for bundle management.

### PR DESCRIPTION
**Pull request by scotje**
work in progress on adding bundle management acceptance tests, extracted from #62

### CODE REVIEW

**Comment by DavidS** [comment]
Rebased on top of master with @rodjek's changes in #65. Let's see if this fares better.

**Comment by DavidS** [comment]
Yey, this now fails on nokogiri 1.8.0 on ruby 2.1, and a mismatch in output on ruby 2.3. I'll call this a success.

File: spec/acceptance/bundle_management_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    describe file("#{target_dir}/Gemfile.lock") do
+      it { is_expected.to be_file }
+    end
+  end
+
+  context 'when there is an invalid Gemfile' do
+    before(:all) do
+      shell_ex("mv #{target_dir}/Gemfile #{target_dir}/Gemfile.old")
```
Comment (scotje) [inline spec/acceptance/bundle_management_spec.rb]: Not sure why it's not reflected in the error output, but I think `mv` is failing on Windows which prevents this test from generating the expected output.

**Comment by scotje** [comment]
`nokogiri` thing seems to be an issue with the pinning in the puppet-module-gems for Ruby 2.1.9. cc @bmjen

File: spec/acceptance/bundle_management_spec.rb (hunk 1 above)
Comment (scotje) [inline spec/acceptance/bundle_management_spec.rb]: Are there platform independent implementations for `rm` and `mv` available yet or do we need to add those?

Comment (DavidS) [inline spec/acceptance/bundle_management_spec.rb]: I had the feeling that all environments we're running in have cygwin or similar installed. Seems like I was wrong. Can we conclusively prove it one way or the other?

Comment (DavidS) [inline spec/acceptance/bundle_management_spec.rb]: re "silent failures", shell_ex returns a `BeakerResult` (or so) that has `exit_code`, `stdout`, and `stderr` members to inspect.

**Comment by scotje** [comment]
I have a semi-permanent Windows dev machine set up in SLICE now so I'll try debugging the 2.3 failures with that.

File: spec/acceptance/bundle_management_spec.rb (hunk 1 above)
Comment (rodjek) [inline spec/acceptance/bundle_management_spec.rb]: I think an important consideration here is also "Do/Can we assume that all our Windows users have a cygwin environment?"

**Comment by DavidS** [comment]
pushed again with no beaker involved. These worked locally on Linux, as well as running against the 0a7b256aba8c7ae56c3867451a581ce409c439c7 puppet-sdk-x64.msi on a Win2012r2 VM.

**Comment by DavidS** [comment]
So caching on appveyor is still broken, even with those changes, but at least we got a green build on appveyor now! 🎉

**Comment by james-stocks** [comment]
The _acceptance:package_ command is now useless (I suspect it will appear to pass, but will just be executing the tests locally again).
I created SDK-275 for the work we want to do in having tests run against a VM

File: appveyor.yml [hunk 2]
```diff
@@ -1,17 +1,24 @@
-build: off
+# version: 1.0.{build}-{branch}
 
-# ruby versions under test
-platform:
-  - Ruby23-x64
-  - Ruby21-x64
+environment:
+  matrix:
+    - RUBY_VERSION: 23
```
Comment (james-stocks) [inline appveyor.yml]: It looks like this changes from running 64bit Ruby to running 32 bit Ruby ( https://www.appveyor.com/docs/build-environment/#ruby )
Does that matter?

**Pull request merged**